### ISSUE TITLE
getPlayer() function not available on first tick

### ISSUE THREAD

**Issue by nearnshaw** [decentraland/creator-hub#264]
In the new engine, the getPlayer() function doesn't return any results on the first frame of the scene.
In the past (before October) this worked well in the new client. Some time around then, several scenes that relied on this stopped working.

getPlayer() should be populated by the time the first tick of the scene runs, so it should be safe to call inside main()

The alternative function getUserData() is async and can work as an alternative, but we have it marked as deprecated
https://docs.decentraland.org/creator/development-guide/sdk7/deprecated-functions/#deprecated-player-data-methods

**Comment by fcolarich** [comment]
I tried several things, but the issue here is twofold, first, that in the documentation it explicitly says:

> 📔 Note: The player data may not be available on the first frame of the scene, depending on load times. You should verify that the data was returned and otherwise attempt again a few milliseconds later.

So, it is not a requisite that this data is there to begin with.

And second, if we want it to work that way, there is the issue that the player data is migrated to the scene from the global world after the scene is ready (of course, cannot be done sooner), which means that it needs the scene to be initialized, but not started to do the migration and then during the scene update cycle on ECS, it creates and puts the player data on the CRDT. 
So with the current implementation we would need to change how we initialize the scene and when we start running the scene simulation, while still running the ECS systems until we are ready to go.

Im attempting several possible solutions, but it might require deeper changes that I wont have the time to do or test properly.

**Comment by nearnshaw** [comment]
@fcolarich 
That note you copied from the docs, I added it recently as a way to mitigate this issue, but it's not how it worked in the old client, and it's not how it worked in Explorer Alpha several months ago either

**Comment by nearnshaw** [comment]
@fcolarich 
I didn't fully understand the concern with the scene lifecycle, as I don't know the Unity side of things

But when we run the main() function in the scene, at that point the scene should be initialized and not yet started. At that point it should have received all the global context data from Unity, and it's only then executing the first tick. Any scene-side systems only then start running, not before.

**Comment by fcolarich** [comment]
Systems coded in the scene, yes, but I meant ECS Unity scene side systems :)
We have ECS systems running for the global world and for the scene world, and those of course only run when the scene is active and update according to the scene FPS. 

It might have been working before, but in the current implementation of the whole scene loading/starting flow it wont work. So we need to refactor it to make it work that way.

Basically, the systems that copy the data from the global world (that has the player profile and all its associated data) to the scene world only do it after the scene is active, which means that during the first frames (even seconds as far as I have been testing), there is no player data. Which is what this ticket is all about xD
We would need to wait until the scene is completely loaded and ready to start, but not started. Then run the systems that copy the data to the scene and only then start updating the scene.

So, we have PlayerCRDTEntitiesHandlerSystem that  tries to add PlayerSceneCRDTEntity to scene player entity when scene is ready, then PlayerProfileDataPropagationSystem that adds the SDKPr﻿ofile to that player entity. These 2 systems run on the global world update cycle and only do these when the scene is ready of course, because we need the scene to be created before it has its own world and systems. (there are more of these that handle the other player data, like wearables and whatnot, but these are the more important ones)
Then we have WritePlayerIdentityDataSystem that puts the profile data onto the player entity on the CRDT and this one runs on the scene world update cycle which only starts after the scene is ready.

So, we need to play with these systems or create new ones while delaying scene start after creation, so we can make sure the data is on the scene already.

TLDR: it needs a refactor to properly do it, while not breaking everything xD

**Comment by mikhail-dcl** [comment]
It's a very old bug and the flow was already reworked. It's worth rechecking @decentraland/qa 
You should be able to create the scene code with a call to the function and visualize it somehow.
